### PR TITLE
add get_bin_data function + tests / namespace

### PR DESCRIPTION
**Pull request by suchprogramming**

### CODE REVIEW

File: lib/also_energy/client.rb [hunk 1]
```diff
@@ -2,15 +2,20 @@
 require 'pry'
 require 'virtus'
 require './lib/also_energy/hash_wrangler'
+require './lib/also_energy/connection'
 require './lib/also_energy/site'
 require './lib/also_energy/hardware'
 
 module AlsoEnergy
+
+  class << self; attr_accessor :session_id; end
```
Comment (dphaener) [inline lib/also_energy/client.rb]: I don't think we can store this as a class variable. Different instances will have different session ids (think multiple customers using the app)

File: lib/also_energy/client.rb [hunk 2]
```diff
@@ -21,7 +26,7 @@ def initialize()
     def login
       message = { 'als:username' => username, 'als:password' => password }
       response = find_in_hash(:session_id, (connection.call(:login, message: message).body))
-      response == nil ? (fail AuthError, "Login Failed!") : (self.session_id = response)
+      response == nil ? (fail AuthError, "Login Failed!") : (self.session_id = response && AlsoEnergy.session_id = response)
```
Comment (dphaener) [inline lib/also_energy/client.rb]: This line is a bit unruly. I would opt for a standard if then block in this case.

File: lib/also_energy/client.rb [hunk 3]
```diff
@@ -36,16 +41,7 @@ def get_site_hardware(site_id)
...
-        globals.env_namespace :soapenv
-        globals.namespace_identifier :als
-        globals.pretty_print_xml true
-        globals.log true
-      end
-    end
   end
 
 end
+binding.pry
```
Comment (dphaener) [inline lib/also_energy/client.rb]: ? Leftover from debugging?

**Comment by suchprogramming** [comment]
sneaky pry!

**Pull request merged**